sdk/mobile-center-distribute/src/test/java/com/microsoft/azure/mobile/distribute/DistributeCustomizationTest.java

```diff
@@ -0,0 +1,367 @@
+        DistributeUtils.getStoredDownloadState();
+        verifyStatic(times(++mobileCenterLogErrorCounter));
+        MobileCenterLog.error(anyString(), anyString());
+
+        /* Verify again to make sure the user action has not been processed yet. */
+        verify(distribute, never()).completeWorkflow();
+    }
+
+    @Test
+    public void handleUserUpdateActionProceededForOptionalUpdate() throws Exception {
+
+        /* Mock http call. */
+        HttpClientNetworkStateHandler httpClient = mock(HttpClientNetworkStateHandler.class);
+        whenNew(HttpClientNetworkStateHandler.class).withAnyArguments().thenReturn(httpClient);
+        when(httpClient.callAsync(anyString(), anyString(), anyMapOf(String.class, String.class), any(HttpClient.CallTemplate.class), any(ServiceCallback.class))).thenAnswer(new Answer<ServiceCall>() {
+
+            @Override
+            public ServiceCall answer(InvocationOnMock invocation) throws Throwable {
+                ((ServiceCallback) invocation.getArguments()[4]).onCallSucceeded("mock");
+                return mock(ServiceCall.class);
+            }
+        });
+
+        /* Mock data model. */
+        mockStatic(ReleaseDetails.class);
+        ReleaseDetails details = mock(ReleaseDetails.class);
+        when(details.getId()).thenReturn(1);
+        when(details.getVersion()).thenReturn(10);
+        when(ReleaseDetails.parse(anyString())).thenReturn(details);
+
+        /* Mock others. */
+        mockStatic(DistributeUtils.class);
+        mockStatic(AsyncTaskUtils.class);
+        when(StorageHelper.PreferencesStorage.getString(PREFERENCE_KEY_UPDATE_TOKEN)).thenReturn("some token");
+
+        /* Mock the download state to DOWNLOAD_STATE_AVAILABLE. */
+        when(DistributeUtils.getStoredDownloadState()).thenReturn(DOWNLOAD_STATE_AVAILABLE);
+
+        /* Set Distribute listener so that Distribute doesn't use default update dialog. */
+        DistributeListener listener = mock(DistributeListener.class);
+        when(listener.onNewReleaseAvailable(eq(mActivity), any(ReleaseDetails.class))).thenReturn(true);
+        Distribute.unsetInstance();
+        Distribute.setListener(listener);
+        Distribute distribute = spy(Distribute.getInstance());
+        doNothing().when(distribute).completeWorkflow();
+
+        /* Start Distribute service. */
+        distribute.onStarted(mActivity, "", mock(Channel.class));
+        distribute.onActivityResumed(mActivity);
+
+        /* Call handleUserUpdateAction. */
+        distribute.handleUserUpdateAction(UserUpdateAction.POSTPONE);
+
+        /* Verify POSTPONE has been processed. */
+        verify(distribute).completeWorkflow();
+
+        /* Call handleUserUpdateAction. */
+        distribute.handleUserUpdateAction(UserUpdateAction.IGNORE);
+
+        /* Verify IGNORE has been processed. */
+        verifyStatic();
+        StorageHelper.PreferencesStorage.putInt(PREFERENCE_KEY_IGNORED_RELEASE_ID, details.getId());
```

Comment:
Yeah.. That makes sense. The previous workflow is completed, it won't allow to work on same release. I'll fix that.